token-metadata/program/tests/mint_new_edition_from_master_edition_via_token.rs

```diff
@@ -423,35 +420,184 @@ mod mint_new_edition_from_master_edition_via_token {
+            .unwrap();
+
+        let master_edition = MasterEditionV2::new(&original_nft);
+        master_edition
+            .create_v3(&mut context, Some(10))
             .await
-            .unwrap()
             .unwrap();
 
-        let master_edition_struct: ProgramMasterEdition =
-            ProgramMasterEdition::safe_deserialize(&master_edition_account.data).unwrap();
+        // Mint the first print edition.
+        let print_edition = EditionMarker::new(&original_nft, &master_edition, 1);
+        print_edition.create(&mut context).await.unwrap();
+
+        let master_edition_struct = master_edition.get_data(&mut context).await;
+        assert!(master_edition_struct.supply == 1);
+        assert!(master_edition_struct.max_supply == Some(10));
+
+        // Try mint edition number 11, this should fail.
+        let print_edition = EditionMarker::new(&original_nft, &master_edition, 11);
+        let err = print_edition.create(&mut context).await.unwrap_err();
+
+        assert_custom_error!(err, MetadataError::EditionNumberGreaterThanMaxSupply);
+
+        // Try mint edition number 999, this should fail.
+        let print_edition = EditionMarker::new(&original_nft, &master_edition, 999);
+        let err = print_edition.create(&mut context).await.unwrap_err();
+
+        assert_custom_error!(err, MetadataError::EditionNumberGreaterThanMaxSupply);
+    }
+
+    #[tokio::test]
+    async fn cannot_remint_existing_edition() {
+        let mut context = program_test().start_with_context().await;
+
+        let original_nft = Metadata::new();
+        original_nft
+            .create_v2(
+                &mut context,
+                "Test".to_string(),
+                "TST".to_string(),
+                "uri".to_string(),
+                None,
+                10,
+                false,
+                None,
+                None,
+                None,
+            )
+            .await
+            .unwrap();
+
+        let master_edition = MasterEditionV2::new(&original_nft);
+        master_edition
+            .create_v3(&mut context, Some(999))
+            .await
+            .unwrap();
+
+        // Mint a couple non-sequential editions.
+        let edition_1 = EditionMarker::new(&original_nft, &master_edition, 1);
+        edition_1.create(&mut context).await.unwrap();
+        let edition_99 = EditionMarker::new(&original_nft, &master_edition, 99);
+        edition_99.create(&mut context).await.unwrap();
+
+        let master_edition_struct = master_edition.get_data(&mut context).await;
+        assert!(master_edition_struct.supply == 2);
+        assert!(master_edition_struct.max_supply == Some(999));
+
+        // Try to remint edition numbers 1 and 99, this should fail.
+        let print_edition = EditionMarker::new(&original_nft, &master_edition, 1);
+        let err = print_edition.create(&mut context).await.unwrap_err();
+
+        assert_custom_error!(err, MetadataError::AlreadyInitialized);
```

Comment:
It is kind of an obscure sounding error message for this case, reused for a couple different things.  A nitpick would be to ask you to add an error code called `EditionTaken` but if you don't want to at the moment that's fine with me.